iceoryx_examples/singleprocess/single_process.cpp

```diff
@@ -102,37 +103,45 @@ void subscriber()
             } while (hasMoreSamples);
         }
 
-        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        std::this_thread::sleep_for(CYCLE_TIME);
     }
+    //! [receive]
 }
 
 int main()
 {
-    // Register sigHandler
-    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
-    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
-
     // set the log level to error to see the essence of the example
+    //! [log level]
     iox::log::LogManager::GetLogManager().SetDefaultLogLevel(iox::log::LogLevel::kError);
+    //! [log level]
 
+    //! [roudi config]
     iox::RouDiConfig_t defaultRouDiConfig = iox::RouDiConfig_t().setDefaults();
     iox::roudi::IceOryxRouDiComponents roudiComponents(defaultRouDiConfig);
+    //! [roudi config]
 
+    //! [roudi]
     iox::roudi::RouDi roudi(roudiComponents.rouDiMemoryManager,
                             roudiComponents.portManager,
                             iox::roudi::RouDi::RoudiStartupParameters{iox::roudi::MonitoringMode::OFF, false});
+    //! [roudi]
 
     // create a single process runtime for inter thread communication
+    //! [runtime]
     iox::runtime::PoshRuntimeSingleProcess runtime("singleProcessDemo");
+    //! [runtime]
 
+    //! [run]
     std::thread publisherThread(publisher), subscriberThread(subscriber);
 
-    // communicate for 2 seconds and then stop the example
-    std::this_thread::sleep_for(std::chrono::seconds(2));
-    keepRunning.store(false);
+    while (!iox::posix::hasTerminationRequested())
+    {
+        std::this_thread::sleep_for(CYCLE_TIME);
+    }
```

Comment:
Why not just `iox::posix::waitForTerminationRequest()`?

Reply:
Because!